client/layout/masterbar/logged-in.jsx

```diff
@@ -191,7 +212,8 @@ export default connect(
 	state => {
 		// Falls back to using the user's primary site if no site has been selected
 		// by the user yet
-		const siteId = getSelectedSiteId( state ) || getPrimarySiteId( state );
+		const currentSelectedSiteId = getSelectedSiteId( state );
+		const siteId = currentSelectedSiteId || getPrimarySiteId( state );
```

Comment:
There's a selector for that FWIW: `getSelectedOrPrimarySiteId`

Reply:
We specifically want to reset only the currently selected site, as trying to reset the primary one might not be the intention of the customer where the primary site and the current site are different.

That's why I extracted the currently selected `siteId` as a separate property.